linera-views/src/backends/lru_caching.rs

```diff
@@ -27,28 +27,74 @@ use crate::{
 use crate::{memory::MemoryStore, store::TestKeyValueStore};
 
 #[cfg(with_metrics)]
-/// The total number of cache faults
-static NUM_CACHE_FAULT: LazyLock<IntCounterVec> =
-    LazyLock::new(|| register_int_counter_vec("num_cache_fault", "Number of cache faults", &[]));
+/// The total number of cache read value faults
+static NUM_CACHE_READ_VALUE_FAULT: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_read_value_fault",
+        "Number of cache read value faults",
+        &[],
+    )
+});
 
 #[cfg(with_metrics)]
-/// The total number of cache successes
-static NUM_CACHE_SUCCESS: LazyLock<IntCounterVec> =
-    LazyLock::new(|| register_int_counter_vec("num_cache_success", "Number of cache success", &[]));
+/// The total number of cache read value successes
+static NUM_CACHE_READ_VALUE_SUCCESS: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_read_value_success",
+        "Number of cache read value success",
+        &[],
+    )
+});
+
+#[cfg(with_metrics)]
+/// The total number of cache contains key faults
+static NUM_CACHE_CONTAINS_KEY_FAULT: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_contains_key_fault",
+        "Number of cache contains key faults",
+        &[],
+    )
+});
+
+#[cfg(with_metrics)]
+/// The total number of cache contains key successes
+static NUM_CACHE_CONTAINS_KEY_SUCCESS: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_contains_key_success",
+        "Number of cache contains key success",
+        &[],
+    )
+});
+
+enum CacheEntry {
+    DoesNotExist,
+    Exists,
+    Value(Vec<u8>),
+}
+
+impl CacheEntry {
+    fn get_contains_key(&self) -> bool {
```

Comment:
One of the uses is negated anyway, so maybe `matches!(entry, CacheEntry::DoesNotExist)` (or even `entry == CacheEntry::DoesNotExist`) would be easier to read, and we don't need this method either.

Reply:
Thank you, done.